Update config parameters (#120)
l3af-project/l3afd: docs/configdoc.md

# L3AFD Config Options Documentation

> See [l3afd.cfg](https://github.com/l3af-project/l3afd/blob/main/config/l3afd.cfg) for a full example configuration.

Reviewer: lets convert this to relative link since it references file in same repo

Reviewer: I think this got missed in your last commit

|BpfMapDefaultPath|`"/sys/fs/bpf"`|The base pin path for eBPF maps| Yes |

## [kf-repo]
| FieldName     | Example       | Description     | Required |
| ------------- | ------------- | --------------- |----------|
|url| `"http://localhost:8000/"`|Default repository from which to download eBPF packages| Yes      |

## [web]

| FieldName     | Example       | Description     | Required |
| ------------- | ------------- | --------------- |----------|
|metrics-addr|`"0.0.0.0:8898"`|Prometheus endpoint for pulling/scraping the metrics.  For more info about Prometheus see [prometheus.io](https://prometheus.io/) | Yes      |
|kf-poll-interval|`"30s"`|Periodic interval at which to scrape metrics using Prometheus| No       |

> |n-metric-samples|`"20"`|Number of Metric Samples| No       |

Reviewer: I think changing example to default makes sense and instead of giving example on each config option here we can point to the complete example config.

|BpfMapDefaultPath|`"/sys/fs/bpf"`|The base pin path for eBPF maps| Yes |

## [kf-repo]

> | FieldName     | Example       | Description     | Required |

Reviewer: Please change heading "Example" to "Default" to match the format below.  We should do this for all sections.

|ingress-map-name| `"tc_ingress_root_array"` |Ingress map name of tc_root program| Yes |
|egress-map-name| `"tc_egress_root_array"`  |Egress map name of tc_root program,for more info about ingress/egress check [cilium](https://docs.cilium.io/en/v1.9/concepts/ebpf/intro/)| Yes |
|command| `"tc_root"`               |Command to run tc_root program| Yes |
|version| `"1.0"`                   |Version of tc_root program| Yes |

## [ebpf-chain-debug]
| FieldName | Default            | Description                                                    | Required |
|-----------|--------------------|----------------------------------------------------------------|----------|
| addr      | `"localhost:8899"` | Hostname and Port of chaining debug REST API                   | No       |
| enabled   | `"true"`           | Boolean to check ebpf chaining debug details is enabled or not | No       |

## [l3af-configs]
| FieldName     | Default       | Description     | Required |
| ------------- | ------------- | --------------- |----------|
|restapi-addr|`"localhost:53000"`| Hostname and Port of l3af-configs REST API | No       |

> # [l3af-config-store]

Reviewer: should be heading2 "##" similar to above

| FieldName     | Default       | Description     | Required        |
| ------------- | ------------- | --------------- | --------------- |
|filename|`"/etc/l3afd/l3af-config.json"`|Absolute path of persistent config file where we are storing L3afBPFPrograms objects. For more info see [models](https://github.com/l3af-project/l3afd/blob/main/models/l3afd.go)| Yes |

> # [mtls]

Reviewer: should be heading2 "##" similar to above

|-----------|--------------------|----------------------------------------------------------------|----------|
| addr      | `"localhost:8899"` | Hostname and Port of chaining debug REST API                   | No       |
| enabled   | `"true"`           | Boolean to check ebpf chaining debug details is enabled or not | No       |

## [l3af-configs]
| FieldName     | Example       | Description     | Required |
| ------------- | ------------- | --------------- |----------|
|restapi-addr|`"localhost:53000"`| Hostname and Port of l3af-configs REST API | No       |

# [l3af-config-store]
| FieldName     | Example       | Description     | Required        |
| ------------- | ------------- | --------------- | --------------- |
|filename|`"/etc/l3afd/l3af-config.json"`|Absolute path of persistent config file where we are storing L3afBPFPrograms objects. For more info see [models](https://github.com/l3af-project/l3afd/blob/main/models/l3afd.go)| Yes |

# [mtls]
| FieldName     | Example       | Description     | Required |
| ------------- | ------------- | --------------- |----------|

> |enabled| `"true"` | Boolean to check mtls enabled or not on REST API exposed by l3afd| No       |

Reviewer 2: ```suggestion
|enabled| `"true"` | Boolean controlling whether mTLS is enabled or not on the REST API exposed by l3afd| No       |
```

|min-tls-version|`"1.3"`| Minimum tls version allowed| No       |

> |cert-dir|`"/etc/l3af/certs"`|Absolute path of ca certificates. On Linux this points to a filesystem directory, but on Windows it can point to [certificate store](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/certificate-stores) | No       |

Reviewer 2: ```suggestion
|cert-dir|`"/etc/l3af/certs"`|Absolute path of CA certificates. On Linux this points to a filesystem directory, but on Windows it can point to a [certificate store](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/certificate-stores) | No       |
```
